18252. ignore sync of items without video_path

paulswartz opened the pull request:
If for some reason one does make it to conversion_for_info(), throw up a soft
failure.

Commit:
18252. ignore sync of items without video_path

If for some reason one does make it to conversion_for_info(), throw up a soft
failure.

bendk commented:
Talked with paul about this one.  It seems good, but it would also be good to try to fix why we are calling conversion_for_info() in the first place.  He's going to work on this one a bit more before we merge it.

Commit:
18252. filter items without a filename from Podcasts

Commit:
18252. using a SQLite field as a boolean doesn't work

sqlite> create table t1(a int, b text);
qlite> insert into t1 values(1, 'foo');
sqlite> insert into t1 values(2, '');
sqlite> insert into t1 values(3, null);
sqlite> select * from t1 where b;
sqlite> select * from t1 where b is not null;
1|foo
2|
sqlite> select * from t1 where b not in (null, "");
1|foo

paulswartz commented:
This is ready for review again.

bendk commented:
Looks great to me.  I think you're the reviewer today, but I guess I'll do the merge.
